firebase-inappmessaging-display/src/main/java/com/google/firebase/inappmessaging/display/internal/FiamWindowManager.java

```diff
@@ -54,6 +54,11 @@ public void show(@NonNull final BindingWrapper bindingWrapper, @NonNull Activity
       return;
     }
 
+    if(activity.isFinishing()){
+      Logging.loge("Activity is finishing or not having valid window token. Cannot show Fiam");
```

Comment:
Activity is finishing or does not have valid window token. Cannot show FIAM

Reply:
Done